### PR TITLE
Allow to replace blocks in a sequentially order following the sequence of matching blocks order

### PR DESCRIPTION
**Pull request by frederico**

### CODE REVIEW

**Review by rchasman — APPROVED** [review]
🙌🏻

File: lib/lm_docstache/parser.rb [hunk 1]
```diff
@@ -47,7 +62,8 @@ class Parser
     #   will be the value that will replace the matched string
     def initialize(document, data, options = {})
       @document = document
-      @data = data.transform_keys(&:to_s)
+      @data = data.transform_keys(&:to_s).select {|e, v| !v.is_a?(Array) }
+      @data_sequential_replacement = data.transform_keys(&:to_s).select {|e, v| v.is_a?(Array) }
```
Comment (leandro) [inline lib/lm_docstache/parser.rb:66]: As an alternative solution, you could have used `Enumerable#partition` in here:
```suggestion
      @data_sequential_replacement, @data =
        data.partition { |_, v| v.is_a?(array) }.map(&:to_h).stringify_keys
```

File: lib/lm_docstache/parser.rb [hunk 2]
```diff
@@ -140,6 +157,32 @@ def replace_variables_in_document!
...
+
+              pattern_found +=1
+              value
+            end
+
+            text_node.content = text
+          end
+        end
+      end
+    end
```
Comment (leandro) [inline lib/lm_docstache/parser.rb:184]: As a suggestion to make things simpler, I suggest the following:
(for instance, we actually don't need the counter)
```suggestion
    def replace_data_sequentially_in_document!
      data_sequential_replacement.each do |tag_key, values|
        values = values.dup
        tag = Regexp.escape("{{#{tag_key}}}")

        document.css('w|t').each do |text_node|
          next unless (text = text_node.text).match(tag)

          text_node.content =  text.gsub(/#{tag}/) { |_| values.any? ? values.shift : '' }
        end
      end
    end
```

**Review by leandro — COMMENTED** [review]
It looked simple and efficient, Fred! 👏🏻 

I left some observations/suggestions, nonetheless.

**Pull request merged**